test/vecelement.jl

```diff
@@ -73,6 +73,13 @@ end
 
 @test isa(VecElement((1,2)), VecElement{Tuple{Int,Int}})
 
+# test for alignment agreement (#32414)
+@noinline function bar32414(a)
+    v = ntuple(w -> Float64(10w), Val(8))
+    return a, (v, (a, (1e6, 1e9)))
+end
+@test bar32414(-35.0) === (-35.0, ((10.0, 20.0, 30.0, 40.0, 50.0, 60.0, 70.0, 80.0), (-35.0, (1.0e6, 1.0e9))))
```

Comment:
Unlike the example in #32414 this test passes for me on master. Should we add a version that includes `VecElement`?

Reply:
oh, I just pasted the wrong one here